### PR TITLE
[TTS] New structure for TTS datasets in scripts/dataset_processing, VocoderDataset, update TTSDataset

### PR DESCRIPTION
**Pull request by Oktai15**
- [x] Create unify downloading and processing TTS datasets in `scripts/dataset_proccesing/tts`
- [x] Add `VocoderDataset`
- [x] Fix problem with collision in name (e.g. LibriTTS suffers) for supplementary data in `TTSDataset`
- [x] Other minor updates in `collection.tts.torch` 

Signed-off-by: Oktai Tatanov <oktai.tatanov@gmail.com>

### CODE REVIEW

File: scripts/dataset_processing/tts/extract_sup_data.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import torch
+from tqdm import tqdm
+from hydra.utils import instantiate
+from nemo.core.config import hydra_runner
+
+
+def preprocess_ds_for_fastpitch_align(dataloader):
```
Comment (blisc) [inline scripts/dataset_processing/tts/extract_sup_data.py:7]: Is the point of this file just to extract pitch statistics?
Should we run normalizer and save the normalized text to a .json? Instead of running everytime we call train, would save a decent amount of time.

File: nemo/collections/tts/torch/readme.md [hunk 2]
```diff
@@ -49,14 +49,3 @@ for batch in tqdm(dataloader, total=len(dataloader)):
 pitch_tensor = torch.cat(pitch_list)
```
Comment (blisc) [inline nemo/collections/tts/torch/readme.md:49]: The get_cmudict.sh script is redundant. We have the entire dict in our repo. Let's just delete that file.

File: nemo/collections/tts/torch/data.py [hunk 3]
```diff
@@ -348,7 +351,10 @@ def __getitem__(self, index):
             if mel_path is not None and Path(mel_path).exists():
                 log_mel = torch.load(mel_path)
             else:
-                mel_path = Path(self.sup_data_path) / f"mel_{audio_stem}.pt"
+                mel_folder = Path(self.sup_data_path) / "mel"
+                mel_folder.mkdir(exist_ok=True, parents=True)
```
Comment (blisc) [inline nemo/collections/tts/torch/data.py:355]: Let's do mkdir in init not in getitem

File: nemo/collections/tts/torch/data.py [hunk 4]
```diff
@@ -369,7 +375,10 @@ def __getitem__(self, index):
                 mel_len = self.get_log_mel(audio).shape[2]
                 align_prior_matrix = torch.from_numpy(self.beta_binomial_interpolator(mel_len, text_length.item()))
             else:
-                prior_path = Path(self.sup_data_path) / f"pr_{audio_stem}.pt"
+                prior_folder = Path(self.sup_data_path) / "align_prior_matrix"
+                prior_folder.mkdir(exist_ok=True, parents=True)
```
Comment (blisc) [inline nemo/collections/tts/torch/data.py:379]: Let's do mkdir in init not in getitem

File: nemo/collections/tts/torch/data.py [hunk 5]
```diff
@@ -333,7 +336,7 @@ def get_log_mel(self, audio):
 
     def __getitem__(self, index):
         sample = self.data[index]
-        audio_stem = Path(sample["audio_filepath"]).stem
+        audio_path_as_text_id = sample["audio_filepath"].replace("/", "-").split(".")[0]
```
Comment (blisc) [inline nemo/collections/tts/torch/data.py:339]: Maybe do audio_stem + deterministic has of the whole path?
Not sure which is better but storing the whole path is pretty extreme.

File: nemo/collections/tts/torch/data.py [hunk 6]
```diff
@@ -381,13 +390,11 @@ def __getitem__(self, index):
 
         pitch, pitch_length = None, None
         if Pitch in self.sup_data_types_set:
-            pitch_name = (
-                f"{audio_stem}_pitch_pyin_"
-                f"fmin{self.pitch_fmin}_fmax{self.pitch_fmax}_"
-                f"fl{self.win_length}_hs{self.hop_len}.pt"
-            )
+            pitch_folder = Path(self.sup_data_path) / "pitch"
+            pitch_folder.mkdir(exist_ok=True, parents=True)
```
Comment (blisc) [inline nemo/collections/tts/torch/data.py:394]: Let's do mkdir in init not in getitem

File: nemo/collections/tts/torch/data.py [hunk 7]
```diff
@@ -411,7 +418,11 @@ def __getitem__(self, index):
 
         energy, energy_length = None, None
         if Energy in self.sup_data_types_set:
-            energy_path = Path(self.sup_data_path) / f"{audio_stem}_energy_wl{self.win_length}_hs{self.hop_len}.pt"
+            energy_folder = Path(self.sup_data_path) / "energy"
+            energy_folder.mkdir(exist_ok=True, parents=True)
```
Comment (blisc) [inline nemo/collections/tts/torch/data.py:422]: Let's do mkdir in init not in getitem

File: nemo/collections/tts/torch/data.py [hunk 8]
```diff
@@ -646,3 +657,148 @@ def _collate_fn(self, batch):
 
         joined_data = self.join_data(data_dict)
         return joined_data
+
+
+class VocoderDataset(Dataset):
```
Comment (blisc) [inline nemo/collections/tts/torch/data.py:662]: Is there no way to make a common subclass for TTSDataset and VocoderDataset?

**Review by blisc — CHANGES_REQUESTED** [review]

File: nemo/collections/tts/torch/data.py (hunk 3 above)
Comment (Oktai15) [reply nemo/collections/tts/torch/data.py:355]: Ok.

File: nemo/collections/tts/torch/data.py (hunk 4 above)
Comment (Oktai15) [reply nemo/collections/tts/torch/data.py:379]: Ok.

File: nemo/collections/tts/torch/data.py (hunk 6 above)
Comment (Oktai15) [reply nemo/collections/tts/torch/data.py:394]: Ok.

File: nemo/collections/tts/torch/data.py (hunk 7 above)
Comment (Oktai15) [reply nemo/collections/tts/torch/data.py:422]: Ok.

File: nemo/collections/tts/torch/data.py (hunk 5 above)
Comment (Oktai15) [reply nemo/collections/tts/torch/data.py:339]: Agree, I thought about it for a long time. Will try to suggest something else.

File: nemo/collections/tts/torch/data.py (hunk 8 above)
Comment (Oktai15) [reply nemo/collections/tts/torch/data.py:662]: There is no chance to create common subclass, but I will try to unify code as mush as possible.

**Review by blisc — APPROVED** [review]
